packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -66,14 +66,13 @@ function GridFilterInputValue(props: GridTypeFilterInputValueProps & TextFieldPr
           .find((optionValue) => String(optionValue) === value);
       }
 
+      if (type === 'number' && value === '') {
+        value = NaN; // Prevents the value parser from converting '' to 0 (zero)
+      }
```

Comment:
> I wonder, it feels like we add custom logic for a specific column type.

Yes, but this problem is bigger than that. `GrildFilterInputValue` already contains custom logic for `singleSelect`. We need to split it into different filter components.

> It feels like we have the same bug with the edit feature

I'm aware of this. However, nobody complained about it so I don't if someone is already treating it as a feature (empty the field to get zero) or everybody is adding a custom value formatter. 🤷‍♀️

Reply:
> GrildFilterInputValue already contains custom logic for singleSelect

It doesn't make it right or justify to do more of it if it's wrong 😁.

> nobody complained about it

Can you envision a case where it could be right? I don't personally. If I empty the input, I want an empty value, not 0. I can input 0 if I want this value.

Reply:
I moved this part to the value parser: https://github.com/mui-org/material-ui-x/pull/3018/files#diff-8e275a43cf9db1cb50555f033653ff9cab146383e1b06aba6f3fbb0ebd365735R13

> Can you envision a case where it could be right? I don't personally. As an end-user, if I empty the input, I want an empty value, not 0. I can input 0 if I want this value.

It could be used to force a number when the user empties the input. If the field doesn't allow null values, defaulting to 0 is something desirable.

Reply:
> Yes, but this problem is bigger than that. `GrildFilterInputValue` already contains custom logic for `singleSelect`. We need to split it into different filter components.

At the time when the `singleSelect` was created it was ok to have it reuse the `GridFilterInputValue` but I agree, it's not ideal and if we introduce `multiSelect` it will become too complex to manage. We can have a separate ticket to track the effort for this tech debt, too much DRY.